src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -91,10 +91,17 @@ class ActionPlanService(
 
   fun approveActionPlan(id: UUID, user: AuthUser): ActionPlan {
     val actionPlan = getActionPlan(id)
+
+    actionPlan.referral.approvedActionPlan?. let {
+      val sessions = deliverySessionRepository.findAllByActionPlanId(id)
```

Comment:
Is the intent here to add on top over the session numbers from the previously approved action plan?
If so, would `id` not need to be `it.id`?

Reply:
Also, if true, naming the variable to be more explicity on it's intent would be nice.

i.e. `sessions` -> `previouslyApprovedSessions`

Reply:
I also think this logic would work better in `DeliverySessionsService` encapsulated in a single method that handles session incrementation so the caller of that class doesn't have to worry about that complexity.